src/Orleans/Orleans.csproj

```diff
@@ -74,8 +74,9 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
-    <Reference Include="Newtonsoft.Json">
-      <HintPath>$(SolutionDir)packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```

Comment:
Don't get me wrong - the objection here is not about "aesthetics" at all. 

The problem is that using fully versioned assembly names in `References` creates "HARD" (4-digit versioned) dependencies rather than ""SOFT" dependencies.

If you blissfully assume that NuGet tool is "doing the right thing" when it demonstrably is not, then you will spend the next several months running in to more places that you also need to then add bindingRedirects to "fix" the brittle version links that NuGet added. 

The versioning problem does not really show itself within the Orleans code base necessarily (except Sample Apps maybe), but very quickly emerges when you start to build more complex multi-component / independently versioned products on top of the Orleans layer(s). This is exactly what happened with the interns on the Orleans-Geo project. :(

If you have a very clear and principled picture of how the packages and software should be built and composed, and make sure that what NuGet follows __that__ model rather than its own heuristics, then things just magically work right and everybody thinks you are a crazy ranting so-and-so because "what versioning problems?" [for the last 5-6 years, BTW]  ;)